src/__tests__/useDidUnmount.ts

```diff
@@ -52,10 +54,38 @@ it('runs an async funciton', async () => {
 
   unmount();
 
-  expect(cleanup.mock.calls.length).toEqual(1);
+  expect(cleanup).toHaveBeenCalledTimes(1);
   expect(finished).toBe(false);
 
   resolve!();
 
   expect(finished).toBe(false);
 });
+
+it('calls the effect with updated dependencies', () => {
+  const secondCleanup = jest.fn();
+  const { unmount, rerender } = renderHook(hook, {
+    initialProps: { func: cleanup, deps: [cleanup] },
+  });
+
+  // updating dependencies
+  rerender({ func: secondCleanup, deps: [secondCleanup] });
+  unmount();
+
+  expect(cleanup).not.toHaveBeenCalled();
+  expect(secondCleanup).toHaveBeenCalledTimes(1);
+});
+
+it('does not update the effect when the dependencies have not been updated', () => {
+  const secondCleanup = jest.fn();
+  const { unmount, rerender } = renderHook(hook, {
+    initialProps: { func: cleanup, deps: [cleanup] },
+  });
+
+  // sending a new cleanup func, but keeping the same dependencies
+  rerender({ func: secondCleanup, deps: [cleanup] });
+  unmount();
```

Comment:
What happens if we call `unmount` one more time here? Does it make sense for us to add a test for this scenario?